LayoutKit/LayoutArrangement.swift

```diff
@@ -82,21 +116,46 @@ public struct LayoutArrangement {
     }
 
     /// Returns the views for the layout and all of its sublayouts.
-    private func makeSubviews() -> [UIView] {
+    private func makeSubviews(from recycler: ViewRecycler, prepareAnimation: Bool) -> [UIView] {
         let subviews = sublayouts.flatMap({ (sublayout: LayoutArrangement) -> [UIView] in
-            return sublayout.makeSubviews()
+            return sublayout.makeSubviews(from: recycler, prepareAnimation: prepareAnimation)
         })
-        if let view = layout.makeView() {
-            view.frame = frame
+        // If we are preparing an animation, then we don't want to update frames or configure views.
+        if let view = layout.makeView(from: recycler, configure: !prepareAnimation) {
+            if !prepareAnimation {
```

Comment:
It is certainly possible to do this in two passes, but I think the code would be more complicated due to view collapsing. It is nice that we can do it incrementally, e.g.:

```
if !prepareAnimation {
    for subview in subviews {
        subview.frame.offsetInPlace(dx: frame.origin.x, dy: frame.origin.y)
    }
}
```